### PR TITLE
added alpine image by default, gcr must be requested by params

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
### What changes were proposed in this pull request?
Adding an alpine image as default artifact output

### Why are the changes needed?
To avoid vulnerabilities related to Ubuntu:bionics that come from the current image produced. An ubuntu:18:08∞

### Does this PR introduce any user-facing change?
Yes. When creating the image flag has to be provided in the case of choosing Google MarketPlace compliant image

### How was this patch tested?
WIP

### CODE REVIEW

**Comment by RayRoestenburg** [comment]
The approach looks good, let's test this and get it merged.

**Comment by franciscolopezsancho** [comment]
I did test it and the `call-record-aggregator` works but I'm not sure the if `cloudflow-patch-spark-mutatingwebhookconfig` and `cloudflow-sparkoperator-webhook-init` have to have some specific output

**Comment by yuchaoran2011** [comment]
@franciscolopezsancho Interesting. Tomorrow I'll try out your PR and verify what's going on with `cloudflow-patch-spark-mutatingwebhookconfig` and `cloudflow-sparkoperator-webhook-init`.

**Comment by franciscolopezsancho** [comment]
:) Thanks @yuchaoran2011

**Comment by franciscolopezsancho** [comment]
btw @yuchaoran2011 I execute the `bootstrap-install-script-2.0.5.sh` with 
```
WEBHOOK_PATCH_JOB_IMAGE="alpine:3.12"
CLOUDFLOW_INSTALLER_IMAGE_TAG="alpine-597-51b7e8b-dirty"```

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/FlinkRunner.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ import FlinkResource._
 import skuber.ResourceSpecification.Subresources
 
 /**
- * Creates the ConfigMap and the Runner resource (a FlinkResource.CR) that define a Spark [[Runner]].
+ * Creates the ConfigMap and the Runner resource (a FlinkResource.CR) that define a Flink [[Runner]].
```
Comment (RayRoestenburg) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/FlinkRunner.scala:32]: 👍

File: installer/build.sbt [hunk 2]
```diff
@@ -110,3 +83,76 @@ lazy val root = Project("cloudflow-installer", file("."))
...
+        runRaw(
+          "wget -q https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz -O - | tar -zxf - --strip-components=1 --exclude kubectl -C /usr/local/bin"
+        )
+      }
+    }
+  }
+
+
+//alpine implementation, currently default one
+lazy val alpineDockerfile: Def.Setting[sbt.Task[sbtdocker.DockerfileBase]] = {
```
Comment (RayRoestenburg) [inline installer/build.sbt:139]:
```suggestion
lazy val ossDockerfile: Def.Setting[sbt.Task[sbtdocker.DockerfileBase]] = {
```

File: installer/build.sbt [hunk 3]
```diff
@@ -110,3 +83,76 @@ lazy val root = Project("cloudflow-installer", file("."))
...
+          )
+    )
+}
+
+def dockerfileDistro(distro: Option[String]): Def.Setting[sbt.Task[sbtdocker.DockerfileBase]] = {
+  if (distro.isDefined && distro.get == "google") googleDockerfile    
+  else alpineDockerfile  
+}
+
+lazy val googleDockerfile: Def.Setting[sbt.Task[sbtdocker.DockerfileBase]] = {
```
Comment (RayRoestenburg) [inline installer/build.sbt:111]:
```suggestion
lazy val gcpMarketplaceDockerfile: Def.Setting[sbt.Task[sbtdocker.DockerfileBase]] = {
```

File: installer/build.sbt [hunk 4]
```diff
@@ -47,37 +48,9 @@ lazy val root = Project("cloudflow-installer", file("."))
...
-        runRaw(
-          "curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv ./kubectl /usr/local/bin/kubectl"
-        )
-        runRaw(
-          "wget -q https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz -O - | tar -zxf - --strip-components=1 --exclude kubectl -C /usr/local/bin"
-        )
-      }
-    },
+    //creates a docker images
+    imageTag(sys.props.get("distro")),
```
Comment (RayRoestenburg) [inline installer/build.sbt:52]:
```suggestion
    imageTag(sys.props.get("dockerfile")),
```

File: installer/build.sbt [hunk 5]
```diff
@@ -47,37 +48,9 @@ lazy val root = Project("cloudflow-installer", file("."))
...
+    dockerfileDistro(sys.props.get("distro")),
```
Comment (RayRoestenburg) [inline installer/build.sbt:53]:
```suggestion
    dockerfileDistro(sys.props.get("dockerfile")),
```

**Comment by franciscolopezsancho** [comment]
All done now @RayRoestenburg . Don't worry about what I said @yuchaoran2011, Ray and I already review it was all good.

File: installer/build.sbt [hunk 6]
```diff
@@ -110,3 +83,76 @@ lazy val root = Project("cloudflow-installer", file("."))
...
+        )
+        runRaw(
+          "wget -q https://github.com/openshift/origin/releases/download/v3.11.0/openshift-origin-client-tools-v3.11.0-0cbc58b-linux-64bit.tar.gz -O - | tar -zxf - --strip-components=1 --exclude kubectl -C /usr/local/bin"
+        )
+      }
+    }
+  }
+
+
+//alpine implementation, currently default one
```
Comment (RayRoestenburg) [inline installer/build.sbt:138]:
```suggestion
```

**Comment by RayRoestenburg** [comment]
Thanks @franciscolopezsancho ! merged.

**Pull request merged**